modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/Tabs.es.js

```diff
@@ -16,19 +16,36 @@ import ClayLayout from '@clayui/layout';
 import ClayTabs from '@clayui/tabs';
 import React from 'react';
 
+import {PagesVisitor} from '../../utils/visitors.es';
 import {EVENT_TYPES} from '../actions/eventTypes.es';
 import {useForm} from '../hooks/useForm.es';
 
 export const Tabs = ({activePage, pages}) => {
 	const dispatch = useForm();
 
+	let isAutocompleteVisible = true;
+
+	const visitor = new PagesVisitor(pages);
+
+	visitor.mapFields((field) => {
+		if (field.fieldName === 'autocomplete' && !field.visible) {
+			isAutocompleteVisible = false;
+		}
+	});
+
```

Comment:
@brunofariasdeo que tal usar `useMemo` no lugar? Ficaria algo como:

```
const isAutocompleteVisible = useMemo(() => {
        const visitor = new PagesVisitor(pages);
    
	visitor.mapFields((field) => {
		if (field.fieldName === 'autocomplete' && !field.visible) {
			return false;
		}

		return true;
	}
}, [pages]);

```

> useMemo will only recompute the memoized value when one of the dependencies has changed. This optimization helps to avoid expensive calculations on every render.